ding/policy/dqn.py

```diff
@@ -197,15 +207,29 @@ def _forward_learn(self, data: Dict[str, Any]) -> Dict[str, Any]:
             q_value, target_q_value, data['action'], target_q_action, data['reward'], data['done'], data['weight']
         )
         value_gamma = data.get('value_gamma')
-        loss, td_error_per_sample = q_nstep_td_error(data_n, self._gamma, nstep=self._nstep, value_gamma=value_gamma)
+        if self._cfg.other.reward_scale.reward_scale_function == 'normal':
+            loss, td_error_per_sample = q_nstep_td_error(data_n, self._gamma, nstep=self._nstep, value_gamma=value_gamma)
+        elif self._cfg.other.reward_scale.reward_scale_function == 'apex':
+            loss, td_error_per_sample = q_nstep_td_error_with_rescale(data_n, self._gamma, nstep=self._nstep, value_gamma=value_gamma)
+        elif self._cfg.other.reward_scale.reward_scale_function == 'errorclip':
+            loss, td_error_per_sample = q_nstep_td_error_with_error_rescale(data_n, self._gamma, nstep=self._nstep, value_gamma=value_gamma)
+        else:
+            loss, td_error_per_sample = q_nstep_td_error(data_n, self._gamma, nstep=self._nstep, value_gamma=value_gamma)
+            
 
         # ====================
         # Q-learning update
         # ====================
         self._optimizer.zero_grad()
         loss.backward()
+        if self._cfg.other.reward_scale.reward_scale_function == 'consgrad':
+            grad_v_norm = self.g_v(data['next_obs'])
```

Comment:
call `g_v` behind td error will lead to gradient mixture, your grad_v here is `grad_v + grad_td`